src/blob-uploader.js

```diff
@@ -96,4 +96,61 @@ blobUploader.prototype.retrieveBlobUrl = function (
     .catch((err) => Promise.reject(new Error('Error retrieving blob url: ' + err)))
 }
 
+blobUploader.prototype.managedUpload = function (
+  blob /*: Blob */,
+  progressFn /* ?:Function */,
+  cancelEventName /* ?:string */
+) /* :Promise<number> */ {
+  if (!blob) {
+    return Promise.reject(new Error('blob argument not provided'))
+  }
+
+  const vars = privateVars.get(this)
+  if (!vars || !vars.hasOwnProperty('uri')) {
+    return Promise.reject(new Error('blobUploader uri not configured'))
+  }
+
+  const request = new Request(vars.uri + 'v1/temporaryCredentials', {
+    method: 'GET',
+    mode: 'cors'
+  })
+
+  return fetch(request)
+    .then((response) => {
+      if (!response.ok) {
+        return Promise.reject(new Error(response.status + ' ' + response.statusText))
+      }
+      return response.json()
+    })
+    .then((apiResponse) => {
+      const S3 = require('aws-sdk/clients/s3')
+      const s3 = new S3({
+        accessKeyId: apiResponse.credentials.AccessKeyId,
+        secretAccessKey: apiResponse.credentials.SecretAccessKey,
+        sessionToken: apiResponse.credentials.SessionToken,
+        region: 'ap-southeast-2'
+      })
+      const params = {
+        Bucket: apiResponse.bucket,
+        Key: apiResponse.id,
+        Body: blob
+      }
+      const managedUpload = s3.upload(params)
+      if (progressFn) {
+        managedUpload.on('httpUploadProgress', (evt) => {
+          progressFn(evt)
+        })
+      }
+      if (cancelEventName) {
+        window.addEventListener(cancelEventName, () => {
+          managedUpload.abort()
+        })
+      }
+      return managedUpload
+        .promise()
+        .then(() => apiResponse.id)
```

Comment:
I think I would prefer this return the `managedUpload` object so that the calling code of `managedUpload()` can run `managedUpload.abort()` themselves. I would prob keep the second parameter to this function (`progressFn`) for convenience. Our docs could just point at S3 `ManagedUpload` docs, or we could copy and paste them into our docs...